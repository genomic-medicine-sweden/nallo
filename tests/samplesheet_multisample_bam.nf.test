nextflow_pipeline {

    name "Test pipeline GENOMICMEDICINESWEDEN_NALLO"
    script "../main.nf"
    tag "PIPELINE"

    test("samplesheet_multisample_bam | --filter_variants_hgnc_ids null --phaser hiphase") {
        tag "samplesheet_multisample_bam"

        when {
            params {
                pipelines_testdata_base_path = 'https://raw.githubusercontent.com/genomic-medicine-sweden/test-datasets/9e0f40af8fb3ab7f8a3daecaeb03a0acda1e464e/'
                input                        = "$projectDir/assets/samplesheet_multisample_bam.csv"
                outdir                       = "$outputDir"
                filter_variants_hgnc_ids     = null
                phaser                       = "hiphase"
                sv_callers                   = "severus,hificnv"
                sv_callers_to_run            = "severus,hificnv,sniffles"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name      = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path      = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            // bam_files: Stable bam files
            def bam_files        = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore_unstable_alignments', include: ['**/*.bam'])
            // Reference (needed for CRAM inspection)
            def fasta = params.pipelines_testdata_base_path + 'reference/hg38.test.fa'
            // cram_files: Stable cram files
            def cram_files       = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore_unstable_alignments', include: ['**/*.cram'])
            // all_vcf_files: All vcf files - can have unstable variants and/or unstable headers
            def all_vcf_files    = getAllFilesFromDir(params.outdir, include: ['**/*.vcf.gz'])
            // stable_vcf_files: Vcf files with stable variants - can be snapshot using variantsMD5
            def stable_vcf_files = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore_unstable_variants', include: ['**/*.vcf.gz'])
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nallo_pipeline_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path,
                    // All bam files
                    bam_files.collect{ file -> [ file.getName(), bam(file.toString(), stringency: 'silent').getReadsMD5() ] },
                    // All cram files
                    cram_files.collect{ file -> [ file.getName(), cram(file.toString(), fasta, stringency: 'silent').getReadsMD5() ] },
                    // All vcf files
                    all_vcf_files.collect{ file -> [ file.getName(), path(file.toString()).vcf.summary ] },
                    // Stable vcf files
                    stable_vcf_files.collect{ file -> [ file.getName(), path(file.toString()).vcf.variantsMD5 ] }
                ).match() }
            )
        }
    }

}
