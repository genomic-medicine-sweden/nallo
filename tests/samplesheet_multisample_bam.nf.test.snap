{
    "samplesheet_multisample_bam | --phaser hiphase": {
        "content": [
            67,
            {
                "ADD_FOUND_IN_TAG": {
                    "bcftools": 1.2,
                    "busybox_awk": "1.36.1"
                },
                "BCFTOOLS_MERGE_PER_CASE": {
                    "bcftools": 1.2
                },
                "BCFTOOLS_MERGE_PER_SAMPLE": {
                    "bcftools": 1.2
                },
                "BCFTOOLS_QUERY": {
                    "bcftools": 1.2
                },
                "BCFTOOLS_REHEADER": {
                    "bcftools": 1.2
                },
                "BEDTOOLS_MERGE": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_SORT": {
                    "bedtools": "2.31.1"
                },
                "CRAMINO": {
                    "cramino": "0.14.5"
                },
                "CREATE_SAMPLES_FILE": {
                    "gawk": "5.1.0"
                },
                "CREATE_SAMPLES_HAPLOTYPES_FILE": {
                    "gawk": "5.1.0"
                },
                "DIPCALL": {
                    "dipcall": 0.3
                },
                "FASTQC": {
                    "fastqc": "0.12.1"
                },
                "GFASTATS_MATERNAL": {
                    "gfastats": "1.3.6"
                },
                "GFASTATS_PATERNAL": {
                    "gfastats": "1.3.6"
                },
                "GUNZIP_FASTA": {
                    "gunzip": 1.1
                },
                "HIFIASM": {
                    "hifiasm": "0.24.0-r702"
                },
                "MINIMAP2_ALIGN": {
                    "minimap2": "2.28-r1209",
                    "samtools": 1.2
                },
                "MINIMAP2_INDEX": {
                    "minimap2": "2.28-r1209"
                },
                "MINIMAP2_INDEX_DIPCALL": {
                    "minimap2": "2.28-r1209"
                },
                "MOSDEPTH": {
                    "mosdepth": "0.3.10"
                },
                "PARAPHASE": {
                    "minimap2": "2.28-r1209",
                    "paraphase": "3.2.1",
                    "samtools": 1.21
                },
                "RELATE_INFER": {
                    "somalier": "0.2.18"
                },
                "RELATE_RELATE": {
                    "somalier": "0.2.18"
                },
                "SAMPLESHEET_PED": {
                    "create_pedigree_file": 1.0,
                    "python": "3.8.3"
                },
                "SAMTOOLS_FAIDX": {
                    "samtools": 1.21
                },
                "SAMTOOLS_FASTQ": {
                    "samtools": 1.21
                },
                "SAMTOOLS_INDEX_DIPCALL": {
                    "samtools": 1.21
                },
                "SAMTOOLS_MERGE": {
                    "samtools": 1.21
                },
                "SEVERUS": {
                    "severus": 1.1
                },
                "SOMALIER_EXTRACT": {
                    "somalier": "0.2.18"
                },
                "SOMALIER_PED_FAMILY": {
                    "create_pedigree_file": 1.0,
                    "python": "3.8.3"
                },
                "SPLITUBAM": {
                    "splitubam": "0.1.1"
                },
                "SPLIT_BED_CHUNKS": {
                    "split_bed_chunks": 1.0
                },
                "SVDB_MERGE": {
                    "svdb": "2.8.2",
                    "bcftools": 1.21
                },
                "TABIX_SVDB_MERGE": {
                    "tabix": 1.2
                },
                "UNTAR_VEP_CACHE": {
                    "untar": 1.34
                },
                "Workflow": {
                    "genomic-medicine-sweden/nallo": "v0.5.0dev"
                }
            },
            [
                "aligned_reads",
                "aligned_reads/HG002_Revio_A",
                "aligned_reads/HG002_Revio_B",
                "assembly_haplotypes",
                "assembly_haplotypes/gfastats",
                "assembly_haplotypes/gfastats/HG002_Revio_A",
                "assembly_haplotypes/gfastats/HG002_Revio_A/HG002_Revio_A.asm.bp.hap1.p_ctg.assembly_summary",
                "assembly_haplotypes/gfastats/HG002_Revio_A/HG002_Revio_A.asm.bp.hap1.p_ctg.fasta.gz",
                "assembly_haplotypes/gfastats/HG002_Revio_A/HG002_Revio_A.asm.bp.hap2.p_ctg.assembly_summary",
                "assembly_haplotypes/gfastats/HG002_Revio_A/HG002_Revio_A.asm.bp.hap2.p_ctg.fasta.gz",
                "assembly_haplotypes/gfastats/HG002_Revio_B",
                "assembly_haplotypes/gfastats/HG002_Revio_B/HG002_Revio_B.asm.bp.hap1.p_ctg.assembly_summary",
                "assembly_haplotypes/gfastats/HG002_Revio_B/HG002_Revio_B.asm.bp.hap1.p_ctg.fasta.gz",
                "assembly_haplotypes/gfastats/HG002_Revio_B/HG002_Revio_B.asm.bp.hap2.p_ctg.assembly_summary",
                "assembly_haplotypes/gfastats/HG002_Revio_B/HG002_Revio_B.asm.bp.hap2.p_ctg.fasta.gz",
                "assembly_variant_calling",
                "assembly_variant_calling/dipcall",
                "assembly_variant_calling/dipcall/HG002_Revio_A",
                "assembly_variant_calling/dipcall/HG002_Revio_A/HG002_Revio_A.dip.bed",
                "assembly_variant_calling/dipcall/HG002_Revio_A/HG002_Revio_A.dip.vcf.gz",
                "assembly_variant_calling/dipcall/HG002_Revio_A/HG002_Revio_A.hap1.bam",
                "assembly_variant_calling/dipcall/HG002_Revio_A/HG002_Revio_A.hap1.bam.bai",
                "assembly_variant_calling/dipcall/HG002_Revio_A/HG002_Revio_A.hap1.bed",
                "assembly_variant_calling/dipcall/HG002_Revio_A/HG002_Revio_A.hap1.paf.gz",
                "assembly_variant_calling/dipcall/HG002_Revio_A/HG002_Revio_A.hap1.sam.gz",
                "assembly_variant_calling/dipcall/HG002_Revio_A/HG002_Revio_A.hap1.var.gz",
                "assembly_variant_calling/dipcall/HG002_Revio_A/HG002_Revio_A.hap2.bam",
                "assembly_variant_calling/dipcall/HG002_Revio_A/HG002_Revio_A.hap2.bam.bai",
                "assembly_variant_calling/dipcall/HG002_Revio_A/HG002_Revio_A.hap2.bed",
                "assembly_variant_calling/dipcall/HG002_Revio_A/HG002_Revio_A.hap2.paf.gz",
                "assembly_variant_calling/dipcall/HG002_Revio_A/HG002_Revio_A.hap2.sam.gz",
                "assembly_variant_calling/dipcall/HG002_Revio_A/HG002_Revio_A.hap2.var.gz",
                "assembly_variant_calling/dipcall/HG002_Revio_A/HG002_Revio_A.pair.vcf.gz",
                "assembly_variant_calling/dipcall/HG002_Revio_B",
                "assembly_variant_calling/dipcall/HG002_Revio_B/HG002_Revio_B.dip.bed",
                "assembly_variant_calling/dipcall/HG002_Revio_B/HG002_Revio_B.dip.vcf.gz",
                "assembly_variant_calling/dipcall/HG002_Revio_B/HG002_Revio_B.hap1.bam",
                "assembly_variant_calling/dipcall/HG002_Revio_B/HG002_Revio_B.hap1.bam.bai",
                "assembly_variant_calling/dipcall/HG002_Revio_B/HG002_Revio_B.hap1.bed",
                "assembly_variant_calling/dipcall/HG002_Revio_B/HG002_Revio_B.hap1.paf.gz",
                "assembly_variant_calling/dipcall/HG002_Revio_B/HG002_Revio_B.hap1.sam.gz",
                "assembly_variant_calling/dipcall/HG002_Revio_B/HG002_Revio_B.hap1.var.gz",
                "assembly_variant_calling/dipcall/HG002_Revio_B/HG002_Revio_B.hap2.bam",
                "assembly_variant_calling/dipcall/HG002_Revio_B/HG002_Revio_B.hap2.bam.bai",
                "assembly_variant_calling/dipcall/HG002_Revio_B/HG002_Revio_B.hap2.bed",
                "assembly_variant_calling/dipcall/HG002_Revio_B/HG002_Revio_B.hap2.paf.gz",
                "assembly_variant_calling/dipcall/HG002_Revio_B/HG002_Revio_B.hap2.sam.gz",
                "assembly_variant_calling/dipcall/HG002_Revio_B/HG002_Revio_B.hap2.var.gz",
                "assembly_variant_calling/dipcall/HG002_Revio_B/HG002_Revio_B.pair.vcf.gz",
                "bcftools",
                "bcftools/HG002_Revio_A.txt",
                "bcftools/HG002_Revio_B.txt",
                "bcftools/HG002_Revio_B_merged.vcf.gz",
                "bcftools/HG002_Revio_B_merged.vcf.gz.tbi",
                "bcftools/HG002_Revio_B_paraphase.vcf.gz",
                "bcftools/HG002_Revio_B_paraphase.vcf.gz.tbi",
                "create",
                "create/HG002_Revio_A_samples_file.txt",
                "create/HG002_Revio_B_samples_file.txt",
                "paraphase",
                "paraphase/HG002_Revio_A",
                "paraphase/HG002_Revio_A/HG002_Revio_A.paraphase.bam",
                "paraphase/HG002_Revio_A/HG002_Revio_A.paraphase.bam.bai",
                "paraphase/HG002_Revio_A/HG002_Revio_A.paraphase.json",
                "paraphase/HG002_Revio_A/HG002_Revio_A_paraphase_vcfs",
                "paraphase/HG002_Revio_A/HG002_Revio_A_paraphase_vcfs/HG002_Revio_A_hba.vcf.gz",
                "paraphase/HG002_Revio_A/HG002_Revio_A_paraphase_vcfs/HG002_Revio_A_hba.vcf.gz.tbi",
                "paraphase/HG002_Revio_B",
                "paraphase/HG002_Revio_B/HG002_Revio_B.paraphase.bam",
                "paraphase/HG002_Revio_B/HG002_Revio_B.paraphase.bam.bai",
                "paraphase/HG002_Revio_B/HG002_Revio_B.paraphase.json",
                "paraphase/HG002_Revio_B/HG002_Revio_B_paraphase_vcfs",
                "paraphase/HG002_Revio_B/HG002_Revio_B_paraphase_vcfs/HG002_Revio_B_hba.vcf.gz",
                "paraphase/HG002_Revio_B/HG002_Revio_B_paraphase_vcfs/HG002_Revio_B_hba.vcf.gz.tbi",
                "paraphase/null_merged.vcf.gz",
                "paraphase/null_merged.vcf.gz.tbi",
                "pedigree",
                "pedigree/family",
                "pedigree/family/FAM.ped",
                "pipeline_info",
                "pipeline_info/nallo_pipeline_software_mqc_versions.yml",
                "qc",
                "qc/cramino",
                "qc/cramino/unphased",
                "qc/cramino/unphased/HG002_Revio_A",
                "qc/cramino/unphased/HG002_Revio_A/HG002_Revio_A_cramino_aligned.arrow",
                "qc/cramino/unphased/HG002_Revio_A/HG002_Revio_A_cramino_aligned.txt",
                "qc/cramino/unphased/HG002_Revio_B",
                "qc/cramino/unphased/HG002_Revio_B/HG002_Revio_B_cramino_aligned.arrow",
                "qc/cramino/unphased/HG002_Revio_B/HG002_Revio_B_cramino_aligned.txt",
                "qc/fastqc",
                "qc/fastqc/HG002_Revio_A",
                "qc/fastqc/HG002_Revio_A/HG002_Revio_A_fastqc.html",
                "qc/fastqc/HG002_Revio_A/HG002_Revio_A_fastqc.zip",
                "qc/fastqc/HG002_Revio_B",
                "qc/fastqc/HG002_Revio_B/HG002_Revio_B_fastqc.html",
                "qc/fastqc/HG002_Revio_B/HG002_Revio_B_fastqc.zip",
                "qc/mosdepth",
                "qc/mosdepth/HG002_Revio_A",
                "qc/mosdepth/HG002_Revio_A/HG002_Revio_A.mosdepth.global.dist.txt",
                "qc/mosdepth/HG002_Revio_A/HG002_Revio_A.mosdepth.region.dist.txt",
                "qc/mosdepth/HG002_Revio_A/HG002_Revio_A.mosdepth.summary.txt",
                "qc/mosdepth/HG002_Revio_A/HG002_Revio_A.per-base.d4",
                "qc/mosdepth/HG002_Revio_A/HG002_Revio_A.regions.bed.gz",
                "qc/mosdepth/HG002_Revio_A/HG002_Revio_A.regions.bed.gz.csi",
                "qc/mosdepth/HG002_Revio_B",
                "qc/mosdepth/HG002_Revio_B/HG002_Revio_B.mosdepth.global.dist.txt",
                "qc/mosdepth/HG002_Revio_B/HG002_Revio_B.mosdepth.region.dist.txt",
                "qc/mosdepth/HG002_Revio_B/HG002_Revio_B.mosdepth.summary.txt",
                "qc/mosdepth/HG002_Revio_B/HG002_Revio_B.per-base.d4",
                "qc/mosdepth/HG002_Revio_B/HG002_Revio_B.regions.bed.gz",
                "qc/mosdepth/HG002_Revio_B/HG002_Revio_B.regions.bed.gz.csi",
                "qc/somalier",
                "qc/somalier/relate",
                "qc/somalier/relate/test",
                "qc/somalier/relate/test/test.html",
                "qc/somalier/relate/test/test.pairs.tsv",
                "qc/somalier/relate/test/test.samples.tsv",
                "repeats",
                "repeats/family",
                "repeats/family/FAM",
                "repeats/family/FAM/FAM_repeats_annotated.vcf.gz",
                "repeats/family/FAM/FAM_repeats_annotated.vcf.gz.tbi",
                "repeats/sample",
                "repeats/sample/HG002_Revio_A",
                "repeats/sample/HG002_Revio_A/HG002_Revio_A_sorted.vcf.gz",
                "repeats/sample/HG002_Revio_A/HG002_Revio_A_sorted.vcf.gz.tbi",
                "repeats/sample/HG002_Revio_A/HG002_Revio_A_spanning_sorted.bam",
                "repeats/sample/HG002_Revio_A/HG002_Revio_A_spanning_sorted.bam.bai",
                "repeats/sample/HG002_Revio_B",
                "repeats/sample/HG002_Revio_B/HG002_Revio_B_sorted.vcf.gz",
                "repeats/sample/HG002_Revio_B/HG002_Revio_B_sorted.vcf.gz.tbi",
                "repeats/sample/HG002_Revio_B/HG002_Revio_B_spanning_sorted.bam",
                "repeats/sample/HG002_Revio_B/HG002_Revio_B_spanning_sorted.bam.bai",
                "snvs",
                "snvs/family",
                "snvs/family/FAM",
                "snvs/family/FAM/FAM_snvs_annotated_ranked.vcf.gz",
                "snvs/family/FAM/FAM_snvs_annotated_ranked.vcf.gz.tbi",
                "snvs/family/FAM/FAM_snvs_annotated_ranked_filtered.vcf.gz",
                "snvs/family/FAM/FAM_snvs_annotated_ranked_filtered.vcf.gz.tbi",
                "snvs/sample",
                "snvs/sample/HG002_Revio_A",
                "snvs/sample/HG002_Revio_A/HG002_Revio_A_snvs_annotated_ranked.vcf.gz",
                "snvs/sample/HG002_Revio_A/HG002_Revio_A_snvs_annotated_ranked.vcf.gz.tbi",
                "snvs/sample/HG002_Revio_B",
                "snvs/sample/HG002_Revio_B/HG002_Revio_B_snvs_annotated_ranked.vcf.gz",
                "snvs/sample/HG002_Revio_B/HG002_Revio_B_snvs_annotated_ranked.vcf.gz.tbi",
                "snvs/stats",
                "snvs/stats/sample",
                "snvs/stats/sample/HG002_Revio_A.vcf.gz.bcftools_stats.txt",
                "snvs/stats/sample/HG002_Revio_B.vcf.gz.bcftools_stats.txt",
                "svs",
                "svs/family",
                "svs/family/FAM",
                "svs/sample",
                "svs/sample/HG002_Revio_A",
                "svs/sample/HG002_Revio_A/HG002_Revio_A_svs.vcf.gz",
                "svs/sample/HG002_Revio_A/HG002_Revio_A_svs.vcf.gz.tbi",
                "svs/sample/HG002_Revio_B",
                "svs/sample/HG002_Revio_B/HG002_Revio_B_svs.vcf.gz",
                "svs/sample/HG002_Revio_B/HG002_Revio_B_svs.vcf.gz.tbi"
            ],
            [
                "HG002_Revio_A.asm.bp.hap1.p_ctg.assembly_summary:md5,b949217515292b5a7735af4ad4bd5b11",
                "HG002_Revio_A.asm.bp.hap1.p_ctg.fasta.gz:md5,380ed173f8d50419a86c229dd81a20c2",
                "HG002_Revio_A.asm.bp.hap2.p_ctg.assembly_summary:md5,c3e50f54e9e2871771d1735c54e57c5e",
                "HG002_Revio_A.asm.bp.hap2.p_ctg.fasta.gz:md5,6ca4cd66b46b2bf7b1ed1ee959fba6ab",
                "HG002_Revio_B.asm.bp.hap1.p_ctg.assembly_summary:md5,7fd907d993bf1140eabc2f3c2116df96",
                "HG002_Revio_B.asm.bp.hap1.p_ctg.fasta.gz:md5,323fd57714246acb37e4e5d55e953429",
                "HG002_Revio_B.asm.bp.hap2.p_ctg.assembly_summary:md5,241c10b531eae6c6bd56d71be2b0806c",
                "HG002_Revio_B.asm.bp.hap2.p_ctg.fasta.gz:md5,6481372dcc00afaea3418749a04ffbbf",
                "HG002_Revio_A.dip.bed:md5,8230ffcc5539eef359ab5128fa60e765",
                "HG002_Revio_A.dip.vcf.gz:md5,d5a5d0620cba624a482d0e48e725e232",
                "HG002_Revio_A.hap1.bam:md5,f5f71f0f37cf23c69981468ed1acc977",
                "HG002_Revio_A.hap1.bam.bai:md5,3e8c67dc7fc05362ff09ecbd9d43c205",
                "HG002_Revio_A.hap1.bed:md5,44dcedabe0edb29af441df1f0b55210a",
                "HG002_Revio_A.hap1.paf.gz:md5,680a7278a2d492ed6ee5c7b19dd5d80f",
                "HG002_Revio_A.hap1.sam.gz:md5,7588cdfcad98f12558e80eca03e22a7f",
                "HG002_Revio_A.hap1.var.gz:md5,bd714e935825d8c665445d93c7d82206",
                "HG002_Revio_A.hap2.bam:md5,490c6e0d6aeef42da8b430a237b390c2",
                "HG002_Revio_A.hap2.bam.bai:md5,04a41250610872ae3bf86108be610883",
                "HG002_Revio_A.hap2.bed:md5,6b4b879d4ebe97f778921252469a029c",
                "HG002_Revio_A.hap2.paf.gz:md5,77ff9e7c1b0ed8454f7a10beba01cbf9",
                "HG002_Revio_A.hap2.sam.gz:md5,981e2e08c4b9b4d1d16410514440c2d6",
                "HG002_Revio_A.hap2.var.gz:md5,9ba9303b30730e419138e177c7d1e0c2",
                "HG002_Revio_A.pair.vcf.gz:md5,4680402dd40f18cc177bd50fb618e6f4",
                "HG002_Revio_B.dip.bed:md5,395e61d926c5f9cbda71f770182a52b6",
                "HG002_Revio_B.dip.vcf.gz:md5,3b3644ef50af7d513e44e09bf043d129",
                "HG002_Revio_B.hap1.bam:md5,c0efe721c19a94dab1037b7a80a8ee05",
                "HG002_Revio_B.hap1.bam.bai:md5,479da74059364a71bd24a0a6b98edf8f",
                "HG002_Revio_B.hap1.bed:md5,8d007624a4935390fb4731817e251cff",
                "HG002_Revio_B.hap1.paf.gz:md5,df23f568c86230a5dea8719e47d233b8",
                "HG002_Revio_B.hap1.sam.gz:md5,36f83a3f60747d41f1472a726d2ddad1",
                "HG002_Revio_B.hap1.var.gz:md5,6234237b2aee91083b6d4c67a91c0d0b",
                "HG002_Revio_B.hap2.bam:md5,2081ccd985b5cf938a258183a296aedf",
                "HG002_Revio_B.hap2.bam.bai:md5,d28e7fd4f409cac6d9fb07781aa2184c",
                "HG002_Revio_B.hap2.bed:md5,f4d086b5254b13a8d7836290d2c1b1cb",
                "HG002_Revio_B.hap2.paf.gz:md5,b1cffc2fa1ed71057acc9cef610a7fc6",
                "HG002_Revio_B.hap2.sam.gz:md5,ecdf9b9d50b54776d0555c71a194af7e",
                "HG002_Revio_B.hap2.var.gz:md5,f09b01fcb43bf64f92ca7ec8a2380c25",
                "HG002_Revio_B.pair.vcf.gz:md5,676492865a9d4765835638d52651af73",
                "HG002_Revio_A.txt:md5,a0e25edabd1af8f8460e4178b75d3d99",
                "HG002_Revio_B.txt:md5,a0e25edabd1af8f8460e4178b75d3d99",
                "HG002_Revio_B_merged.vcf.gz:md5,63dfa194d6eebe89c42d7cb4662bb5f9",
                "HG002_Revio_B_merged.vcf.gz.tbi:md5,bef85983779173b3edbd79429bba6062",
                "HG002_Revio_B_paraphase.vcf.gz:md5,a40f15d3f40bad6feab75ae3ca0ba531",
                "HG002_Revio_B_paraphase.vcf.gz.tbi:md5,756640d819c3341590792b82cc237e74",
                "HG002_Revio_A_samples_file.txt:md5,c1acc4c856e1723e296679b85d776892",
                "HG002_Revio_B_samples_file.txt:md5,971e1cc506e698e1174cb30aa601a004",
                "null_merged.vcf.gz:md5,a9fe5244695ab38b39b941c889b74c89",
                "null_merged.vcf.gz.tbi:md5,b8c4d65a06a29e05ae79770da30c6c6d",
                "FAM.ped:md5,24d8694d580f782ed77d4d1b5c6f6fb4",
                "HG002_Revio_A_cramino_aligned.arrow:md5,72df2934ff8aa7e1bf8cf8a4881a0d2a",
                "HG002_Revio_B_cramino_aligned.arrow:md5,544a6513e7604d6425a664855212eced",
                "HG002_Revio_A.mosdepth.global.dist.txt:md5,6186315d4d65eda85553af82a98829d1",
                "HG002_Revio_A.mosdepth.region.dist.txt:md5,c6c7ee8f056b8b2b92c97ec472b1db16",
                "HG002_Revio_A.mosdepth.summary.txt:md5,35c51f1ad9d2856d1e6446205f19a8e3",
                "HG002_Revio_A.per-base.d4:md5,a4341498b23fdd2b81023499b8c4e1aa",
                "HG002_Revio_A.regions.bed.gz:md5,e051f64c7a780d67ea6727a327dd4281",
                "HG002_Revio_A.regions.bed.gz.csi:md5,98ab5f668e864129054ffc1663a980ee",
                "HG002_Revio_B.mosdepth.global.dist.txt:md5,830a8b737fea939dcfea4d1626e60576",
                "HG002_Revio_B.mosdepth.region.dist.txt:md5,d4cb66ac28dc6816869a73757226f902",
                "HG002_Revio_B.mosdepth.summary.txt:md5,191877ae5228e7880276585557a7b218",
                "HG002_Revio_B.per-base.d4:md5,e22533485912a648cc708a4851fa4c7d",
                "HG002_Revio_B.regions.bed.gz:md5,9c4c037139c752c60385d3c0b905af7f",
                "HG002_Revio_B.regions.bed.gz.csi:md5,78d70be812a3d19360a3d51a5fb19432"
            ],
            [
                [
                    "HG002_Revio_A.hap1.bam",
                    "da103c401cf854a00aa75a57cda0f34d"
                ],
                [
                    "HG002_Revio_A.hap2.bam",
                    "e8599cef23fc88e2fe09d2bda5ea372f"
                ],
                [
                    "HG002_Revio_B.hap1.bam",
                    "2352f7a9960c400d8182ad2ec442887"
                ],
                [
                    "HG002_Revio_B.hap2.bam",
                    "6e4f7fa25e9905d538ecc65f923806"
                ],
                [
                    "HG002_Revio_A.paraphase.bam",
                    "12b7845566fc2863bfd5328a5bcbfab6"
                ],
                [
                    "HG002_Revio_B.paraphase.bam",
                    "12b7845566fc2863bfd5328a5bcbfab6"
                ]
            ],
            [
                [
                    "HG002_Revio_A.dip.vcf.gz",
                    "VcfFile [chromosomes=[chrX, chr16], sampleCount=1, variantCount=1917, phased=true, phasedAutodetect=false]"
                ],
                [
                    "HG002_Revio_A.pair.vcf.gz",
                    "VcfFile [chromosomes=[chrX, chr16], sampleCount=2, variantCount=1917, phased=false, phasedAutodetect=false]"
                ],
                [
                    "HG002_Revio_B.dip.vcf.gz",
                    "VcfFile [chromosomes=[chrX, chr16], sampleCount=1, variantCount=1862, phased=true, phasedAutodetect=false]"
                ],
                [
                    "HG002_Revio_B.pair.vcf.gz",
                    "VcfFile [chromosomes=[chrX, chr16], sampleCount=2, variantCount=1862, phased=false, phasedAutodetect=false]"
                ],
                [
                    "HG002_Revio_B_merged.vcf.gz",
                    "VcfFile [chromosomes=[chr16], sampleCount=5, variantCount=3970, phased=true, phasedAutodetect=false]"
                ],
                [
                    "HG002_Revio_B_paraphase.vcf.gz",
                    "VcfFile [chromosomes=[chr16], sampleCount=5, variantCount=3970, phased=true, phasedAutodetect=false]"
                ],
                [
                    "HG002_Revio_A_hba.vcf.gz",
                    "VcfFile [chromosomes=[chr16], sampleCount=5, variantCount=3970, phased=true, phasedAutodetect=false]"
                ],
                [
                    "HG002_Revio_B_hba.vcf.gz",
                    "VcfFile [chromosomes=[chr16], sampleCount=5, variantCount=3970, phased=true, phasedAutodetect=false]"
                ],
                [
                    "HG002_Revio_A_phased.vcf.gz",
                    "VcfFile [chromosomes=[chrX, chr16, chr20], sampleCount=1, variantCount=191, phased=false, phasedAutodetect=false]"
                ],
                [
                    "HG002_Revio_B_phased.vcf.gz",
                    "VcfFile [chromosomes=[chrX, chr16, chr20], sampleCount=1, variantCount=191, phased=false, phasedAutodetect=false]"
                ],
                [
                    "FAM_repeats_annotated.vcf.gz",
                    "VcfFile [chromosomes=[chr20], sampleCount=2, variantCount=2, phased=false, phasedAutodetect=false]"
                ],
                [
                    "HG002_Revio_A_sorted.vcf.gz",
                    "VcfFile [chromosomes=[chr20], sampleCount=1, variantCount=1, phased=false, phasedAutodetect=false]"
                ],
                [
                    "HG002_Revio_B_sorted.vcf.gz",
                    "VcfFile [chromosomes=[chr20], sampleCount=1, variantCount=1, phased=false, phasedAutodetect=false]"
                ],
                [
                    "FAM_snvs_annotated_ranked.vcf.gz",
                    "VcfFile [chromosomes=[chrX, chr16], sampleCount=2, variantCount=114, phased=false, phasedAutodetect=false]"
                ],
                [
                    "FAM_snvs_annotated_ranked_filtered.vcf.gz",
                    "VcfFile [chromosomes=[chr16], sampleCount=2, variantCount=65, phased=false, phasedAutodetect=false]"
                ],
                [
                    "HG002_Revio_A_snvs_annotated_ranked.vcf.gz",
                    "VcfFile [chromosomes=[chrX, chr16], sampleCount=1, variantCount=113, phased=false, phasedAutodetect=false]"
                ],
                [
                    "HG002_Revio_B_snvs_annotated_ranked.vcf.gz",
                    "VcfFile [chromosomes=[chrX, chr16], sampleCount=1, variantCount=112, phased=false, phasedAutodetect=false]"
                ],
                [
                    "FAM_svs_cnvs_merged_annotated_ranked.vcf.gz",
                    "VcfFile [chromosomes=[chrX, chr16], sampleCount=2, variantCount=87, phased=false, phasedAutodetect=false]"
                ],
                [
                    "FAM_svs_cnvs_merged_annotated_ranked_filtered.vcf.gz",
                    "VcfFile [chromosomes=[chr16], sampleCount=2, variantCount=5, phased=false, phasedAutodetect=false]"
                ],
                [
                    "HG002_Revio_A_cnvs.vcf.gz",
                    "VcfFile [chromosomes=[chrX, chr16, chr20], sampleCount=1, variantCount=36, phased=false, phasedAutodetect=false]"
                ],
                [
                    "HG002_Revio_A_svs.vcf.gz",
                    "VcfFile [chromosomes=[chrX, chr16], sampleCount=1, variantCount=55, phased=false, phasedAutodetect=false]"
                ],
                [
                    "HG002_Revio_B_svs.vcf.gz",
                    "VcfFile [chromosomes=[chrX, chr16], sampleCount=1, variantCount=55, phased=false, phasedAutodetect=false]"
                ]
            ],
            [
                [
                    "HG002_Revio_A.dip.vcf.gz",
                    "db10cdb233cb09a46c1694536c2e79e7"
                ],
                [
                    "HG002_Revio_A.pair.vcf.gz",
                    "fa3bfacdab5449526d29d642340d6708"
                ],
                [
                    "HG002_Revio_B.dip.vcf.gz",
                    "eecfa7a0111b2ababb2e37fb91686b3f"
                ],
                [
                    "HG002_Revio_B.pair.vcf.gz",
                    "83dc2bb5d8f46bf3daf99f27e4d3b022"
                ],
                [
                    "HG002_Revio_B_merged.vcf.gz",
                    "2420fa50536f5c9982a9fdc88f31590"
                ],
                [
                    "HG002_Revio_B_paraphase.vcf.gz",
                    "99a7cda0c03a0bd863da3c859a83b6e8"
                ],
                [
                    "null_merged.vcf.gz",
                    "2420fa50536f5c9982a9fdc88f31590"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-11T12:26:22.980472"
    }
}
